userspace/libscap/engine/savefile/scap_reader_gzfile.c

```diff
@@ -26,7 +26,22 @@ typedef struct reader_handle
 static int gzfile_read(scap_reader_t *r, void* buf, uint32_t len)
 {
     ASSERT(r != NULL);
-    return gzread(((reader_handle_t*)r->handle)->m_file, buf, len);
+    int readsize = gzread(((reader_handle_t*)r->handle)->m_file, buf, len);
+
+    if (readsize < (int)len)
```

Comment:
Can't all of this be handled by the client program?

Reply:
I don't think so, at least with the current API. I'm trying to use sinsp::open_savefile() + sinsp::next() to read events from a file which is simultaneously being written by another process. The problem is that EOF handling in zlib is sticky; it changes its internal state when it reaches the end of input which in turn causes subsequent sinsp::next() calls to fail with SCAP_EOF even after more data has been written to the file.

The [gzread documentation](https://github.com/madler/zlib/blob/develop/zlib.h#L1398) recommends using gzclearerr in this case. I don't think I can do so from the client side because we don't appear expose the gzFile handle.

Reply:
Thanks for the clarification, LGTM!